src/controller/java/DeviceAttestationDelegateBridge.cpp

```diff
@@ -0,0 +1,124 @@
+
+    mResult = attestationResult;
+    if (mDeviceAttestationDelegate != nullptr)
+    {
+        JNIEnv * env                 = JniReferences::GetInstance().GetEnvForCurrentThread();
+        jclass completionCallbackCls = nullptr;
+        JniReferences::GetInstance().GetClassRef(
+            env, "chip/devicecontroller/DeviceAttestationDelegate$DeviceAttestationCompletionCallback", completionCallbackCls);
+        VerifyOrReturn(completionCallbackCls != nullptr,
+                       ChipLogError(Controller, "Could not find device attestation completion callback class."));
+        jclass failureCallbackCls = nullptr;
+        JniReferences::GetInstance().GetClassRef(
+            env, "chip/devicecontroller/DeviceAttestationDelegate$DeviceAttestationFailureCallback", failureCallbackCls);
+        VerifyOrReturn(failureCallbackCls != nullptr,
+                       ChipLogError(Controller, "Could not find device attestation failure callback class."));
+
+        if (env->IsInstanceOf(mDeviceAttestationDelegate, completionCallbackCls))
+        {
+            jmethodID onDeviceAttestationCompletedMethod;
+            JniReferences::GetInstance().FindMethod(env, mDeviceAttestationDelegate, "onDeviceAttestationCompleted",
+                                                    "(JLchip/devicecontroller/AttestationInfo;I)V",
+                                                    &onDeviceAttestationCompletedMethod);
+            VerifyOrReturn(onDeviceAttestationCompletedMethod != nullptr,
+                           ChipLogError(Controller, "Could not find deviceAttestation completed method"));
+            jobject javaAttestationInfo;
+            CHIP_ERROR err = N2J_AttestationInfo(env, info, javaAttestationInfo);
+            VerifyOrReturn(err == CHIP_NO_ERROR,
+                           ChipLogError(Controller, "Failed to create AttestationInfo, error: %s", err.AsString()));
+            env->CallVoidMethod(mDeviceAttestationDelegate, onDeviceAttestationCompletedMethod, reinterpret_cast<jlong>(device),
+                                javaAttestationInfo, static_cast<jint>(attestationResult));
+        }
+        else if ((attestationResult != chip::Credentials::AttestationVerificationResult::kSuccess) &&
+                 env->IsInstanceOf(mDeviceAttestationDelegate, failureCallbackCls))
+        {
+            jmethodID onDeviceAttestationFailedMethod;
+            JniReferences::GetInstance().FindMethod(env, mDeviceAttestationDelegate, "onDeviceAttestationFailed", "(JI)V",
+                                                    &onDeviceAttestationFailedMethod);
+            VerifyOrReturn(onDeviceAttestationFailedMethod != nullptr,
+                           ChipLogError(Controller, "Could not find deviceAttestation failed method"));
+            env->CallVoidMethod(mDeviceAttestationDelegate, onDeviceAttestationFailedMethod, reinterpret_cast<jlong>(device),
```

Comment:
onDeviceAttestationFailed is also get called in case the user provided certs failed, right?

Reply:
Yes, already tested in Tuya App, we recently switched our test PAA to an online PAA